### PR TITLE
fix: Allow Team Event editing by the person who created it

### LINKED ISSUE #2035: [CAL-1682] Team members can create team events but can't update them
### Issue Summary

Being a member of a team allows you to create new event types from the team, but doesn't allow you to update them.

### Steps to Reproduce

1. Went to dashboard
2. Created a new event type for a team I'm a member of (not owner)
3. Can't update that same event type

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

### Technical details

- https://app.birdeatsbug.com/sessions/As4F1pYE88T63MtT0LAZa

<sub>[CAL-1682](https://linear.app/calcom/issue/CAL-1682/team-members-can-create-team-events-but-cant-update-them)</sub>

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?
- Allow editing/deleting of a team event by the member who created it.
- Owner/Admin would still have the editing/deleting access for all team events, no matter who created it.

 **This is applicable to all newly created events only because team events are still not having `userId` set**

Fixes #2035

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?
- [x] Created a team event(Event1) and try to update/delete it. It should be allowed. Check that **Readonly** tag goes away in Event Types Listing
- [x] Login as Team owner, you should be able to update/delete the same event(Event1). Check that  **Readonly** tag isn't therein Event Types Listing
- [x] Login as any other member of the team, you should be able to see event Event1. It should have a  **Readonly** tag and you shouldn't be able to update/delete the event. Backend would reject the request.
- [x] Check that non-team event is also editable by the creator.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/prisma/migrations/20220305052238_adds_event_creator/migration.sql [hunk 1]
```diff
@@ -0,0 +1,47 @@
+-- AlterTable
+ALTER TABLE "EventType" ADD COLUMN     "creatorId" INTEGER;
+
+-- AddForeignKey
+ALTER TABLE "EventType" ADD CONSTRAINT "EventType_creatorId_fkey" FOREIGN KEY ("creatorId") REFERENCES "users"("id") ON DELETE SET NULL ON UPDATE CASCADE;
+
+-- RenameIndex
+ALTER INDEX "Booking.uid_unique" RENAME TO "Booking_uid_key";
```
Comment (hariombalhara) [inline packages/prisma/migrations/20220305052238_adds_event_creator/migration.sql:8]: Not sure why these indices renaming happened. This is automatically generated

**Review by pumfleet — COMMENTED** [review]
Upon testing, this seems to work fine. However I did not test what happens when this migration is run on top of existing event types, which will not have a creator ID set. Does this prevent everyone except admins from editing it?

File: apps/web/next.config.js [hunk 2]
```diff
@@ -18,6 +18,7 @@ if (!process.env.NEXT_PUBLIC_APP_URL) {
   process.env.NEXT_PUBLIC_APP_URL = process.env.BASE_URL;
 }
 process.env.NEXT_PUBLIC_BASE_URL = process.env.BASE_URL;
+console.warn("VARS", process.env.PLAYWRIGHT_SECRET);
```
[the remark this answers is not in the document]
Comment (agustif) [reply apps/web/next.config.js:21]: Maybe we could copy userId -> creatorId on the migration, not sure if it would bring some edge cases like events not created by current owner maybe?

**Comment by agustif** [comment]
> Upon testing, this seems to work fine. However I did not test what happens when this migration is run on top of existing event types, which will not have a creator ID set. Does this prevent everyone except admins from editing it?

Maybe we could copy userId -> creatorId on the migration, not sure if it would bring some edge cases like events not created by current owner maybe?

The rest looks good!

File: apps/web/pages/api/auth/forgot-password.ts [hunk 3]
```diff
@@ -63,7 +63,7 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     };
 
     await sendPasswordResetEmail(passwordEmail);
-
+    console.log("FORGOT PASSWORD", process.env.PLAYWRIGHT_SECRET);
```
[the remark this answers is not in the document]
Comment (agustif) [reply apps/web/pages/api/auth/forgot-password.ts:66]: We could add some no-console eslint rules, I always have to ctrl-f and double check before committing lol

**Review by agustif — CHANGES_REQUESTED** [review]
Any idea why the zod/webhook empty file is on the file changes?

image
I think @zomars fixed the zod auto-generated files being picked up.

**Comment by hariombalhara** [comment]
> > Upon testing, this seems to work fine. However I did not test what happens when this migration is run on top of existing event types, which will not have a creator ID set. Does this prevent everyone except admins from editing it?
> 
> Maybe we could copy userId -> creatorId on the migration, not sure if it would bring some edge cases like events not created by current owner maybe?
> 
> The rest looks good!

As per my understanding userId is not set now. Only `users` is set and that actually has a list of attendees for the event type. So, there is no information in PG that can reliably tell us the creator of the event.

I have removed accidental code changes and this PR is ready for review again

**Comment by agustif** [comment]
Hey @hariombalhara code looks good to me. 
Cant we gent the event type owner from userId which lives in db in user event type relation table?

From schema.prisma

relation("user_eventtype")
  userId                  Int?

**Review by agustif — COMMENTED** [review]
LGTM, but I think userId should be available in eventType and might be === to new creatorId? Maybe we can migrate the data from there

**Comment by hariombalhara** [comment]
@agustif regarding that userId thing, I created a new event type and checked DB studio and this is what I saw
Notice, the last one USERIDTEST. It doesn't have `userId` set. It has users though.

image

**Comment by zomars** [comment]
@hariombalhara @agustif Currently we have a many-to-many relationship from users and event types. Hence why we don't have `userId` set. I think this is being used for team events? I could be wrong tho. Tagging @emrysal to see if he knows more about this data model.

**Comment by hariombalhara** [comment]
@zomars This has been fixed recently in https://github.com/calcom/cal.com/pull/2599

Now I can see userId, I would change the code to use that instead of creatorId
Screenshot 2022-05-17 at 3 48 41 PM

**Comment by hariombalhara** [comment]
Updated the PR as now userId is being set. It was missing for team event, fixed that as well. I am not sure why specific check for team event was added earlier.

**Comment by zomars** [comment]
> Updated the PR as now userId is being set. It was missing for team event, fixed that as well. I am not sure why specific check for team event was added earlier.


Does this means that this fix won’t apply to already created records?
